src/main/java/org/opensearch/flowframework/model/ResourcesCreated.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.model;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.flowframework.common.CommonValue.RESOURCE_ID_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_STEP_NAME;
+
+/**
+ * This represents an object in the WorkflowState {@link WorkflowState}.
+ */
+public class ResourcesCreated implements ToXContentObject, Writeable {
+
+    private String workflowStepName;
+    private String resourceId;
```

Comment:
How are we handling the case where multiple resources may be created by a workflow step?  

I'm not sure how a `String` gives me enough information to identify the resource created. Was it an index? A connector? A model?   Or is this JSON content with more detail (in which case naming it 'id' isn't right.

Reply:
Like you mentioned below, we have a list of resources created, we can add multiple of them to workflowData and index if a step requires it. Regarding info to identify the resource created, lets move that to a different PR as we are discussing it on a separate issue and i want to merge this API.